meter/powerwall.go

```diff
@@ -149,3 +207,31 @@ func (m *PowerWall) batterySoc() (float64, error) {
 
 	return float64(res.Percentage), err
 }
+
+// SetBatteryMode implements the api.BatteryController interface
+func (m *PowerWall) SetBatteryMode(mode api.BatteryMode) error {
+	if !m.batteryControl {
+		return nil
+	}
+
+	switch mode {
+	case api.BatteryNormal: // set minSoc to cofigured default
+		if err := m.energySite.SetBatteryReserve(uint64(m.defaultBatteryReserve)); err != nil {
+			return err
+		}
+	case api.BatteryHold: // set minSoc to currentSoc
+		ess, err := m.energySite.EnergySiteStatus()
+		if err != nil {
+			return err
+		}
+		currentSoc := math.Round(ess.PercentageCharged + 0.5) // .5 ensures we round up
+		if err := m.energySite.SetBatteryReserve(uint64(currentSoc)); err != nil {
+			return err
+		}
+	case api.BatteryCharge: // set mincSoc to 100; will not cause grid charge, but any PV surplus will charge
```

Comment:
Wenn das nicht zu Ladung führt gehört hier egtl. `ErrNotAvailable` hin

Reply:
Führt bei mir bei deaktiviertem Grid Charge zumindest zu Ladung von PV Überschuss. Der installierende Betrieb kann ggf. auch Grid Charge aktivieren.

Reply:
Das würde dann in diesem Modus passieren falls aktiviert? Dann passt es. 

> Führt bei mir bei deaktiviertem Grid Charge zumindest zu Ladung von PV Überschuss. 

Das wäre dann Normal

Reply:
Normal würde den PV Überschuss nicht in die Batterie leiten (weil minSoc auf der Hausbatterie erreicht), sondern zur Autoladung beisteuern.
Bei charge würde der PV Überschuss in die Batterie geleitet (zumindest solange wie minSoc (95) nicht erreicht), und das Auto/Gebäude wird rein über Netzstrom versorgt.
Ich denke es passt so.